apps/greenhouse/src/components/AuthProvider.jsx

```diff
@@ -0,0 +1,137 @@
+import React, { createContext, useContext, useState, useMemo, useCallback, useRef } from "react"
+import { oidcSession, mockedSession } from "@cloudoperators/juno-oauth"
+
+const setOrganizationToUrl = (groups) => {
+  const orgString = groups?.find((g) => g.indexOf("organization:") === 0)
+  if (orgString) {
+    const name = orgString.split(":")[1]
+    let url = new URL(window.location.href)
+    url.searchParams.set("org", name)
+    window.history.replaceState(null, null, url.href)
+  }
+}
+
+const AuthContext = createContext()
+
+export const AuthProvider = ({ options, children }) => {
+  const [authData, setAuthData] = useState(null)
+  const [isDemoMode, setIsDemoMode] = useState(false)
+  const oidcRef = useRef(null)
+
+  const initializeOidc = useCallback(() => {
```

Comment:
I believe `useCallback` is unnecessary in this case since it only depends on options, which are provided via props. The component will re-render whenever the props change, so the function will be recreated naturally. Removing useCallback should not affect functionality and will simplify the code.

Reply:
done